source/opt/const_folding_rules.cpp

```diff
@@ -342,6 +342,8 @@ UnaryScalarFoldingRule FoldIToFOp() {
     assert(result_type == a->type() && result_type == b->type());         \
     const analysis::Float* float_type = result_type->AsFloat();           \
     assert(float_type != nullptr);                                        \
+    if (float_type->width() != 32 && float_type->width() != 64)           \
+      return nullptr;                                                     \
```

Comment:
Isn't this already handled? There are two `if` below, and it neither of them are true, we return nullptr.

Reply:
Good point, guess I misread those.